src/TTextEdit.cpp

```diff
@@ -1390,6 +1390,8 @@ QString TTextEdit::getSelectedText(char newlineChar)
         // we never append the last character of a buffer line se we set our own
         text.append(newlineChar);
     }
+    Q_UNREACHABLE();
```

Comment:
Sorry, but I was not sure __this__ is an unreachable point - do you have any insight either way - that is why I suggested a QDebug report - something like:

    qDebug() << "TTextEdit::getSelectedText(...) INFO - unexpectedly hit bottom of method so returning:" << text;

Reply:
I see what you mean. If something goes wrong in the selection process, it would be possible to hit the end of the function. I've removed q_unreachable, and set a debug message instead.

Reply:
😕 Um, I believe you've, to use a highly technical phrase, got it *arse-about-face*!   You have put the `Q_UNREACHABLE` bit where the `qDebug()` should be and vice-versa. {I generally prefix any qDebug or similar debug output with the class and method where it is located - so the `"TTextEdit::getSelectedText(...) ...  "` may have suggested where it ought to be. 😆 }!

Don't worry that this is taking several commits to get the kinks out - when it comes to merging it into the code we'll use the alternate "Squash and merge" approach (our normal is the ¨Rebase and merge") - which will hide all the revisions into a single commit into the development code and all the things that get done and then undone will never be seen outside this PR page... 😈

Reply:
Oops. Total brain fart there. well, should be correct now, and actually I've been using git --amend, so we should still be on a single commit anyway.